### PR TITLE
Update the information already existing instead of fetching everything again

### PR DESCRIPTION
**Pull request by andreeanica16**
1. Add in JobStoreCenter function convertEntityToJobJSON that takes an entity and returns a JobJSON with all the fields written accordingly --> helps with modularizing the programm.
2. Add in JobStoreCenter the possibility to fetch the project from Datastore
3. Add in JobStoreCenter updateJobtoDatastore that only changes the fields of a job that have been modified. If a field has't been modified, its value its null. Only metric fields and state can be modified, so these are the only fields investigated
4. Add in JobStoreCenter method dealWIthProject which decides if the project should be updated (if it already exists) or it should be created.
5. Add in JobModel fetchJobsforUpdate that returns a list with updated jobs (jobs that have unmodified fields null)

### CODE REVIEW

**Comment by andreeanica16** [comment]
@iht @juliocc @tblanshard

File: src/main/java/com/google/sps/JobStoreCenter.java [hunk 1]
```diff
@@ -109,37 +112,159 @@ private void addJobToDatastore(RunningJob job, Entity project) {
...
+    String sdk = (String) entity.getProperty("sdk");
+    String sdkSupportStatus = (String) entity.getProperty("sdkSupportStatus");
+    String region = (String) entity.getProperty("region");
+    int currentWorkers = ((Long) entity.getProperty("currentWorkers")).intValue();
+    String startTime = (String) entity.getProperty("startTime");
+    Double totalVCPUTime = (Double) entity.getProperty("totalVCPUTime"); 
+    Double totalMemoryTime = (Double) entity.getProperty("totalMemoryTime"); 
+    Double totalDiskTimeHDD = (Double) entity.getProperty("totalDiskTimeHDD");
+    Double totalDiskTimeSSD = (Double) entity.getProperty("totalDiskTimeSSD");
+    Long longCurrentVcouCount = (Long) entity.getProperty("currentVcpuCount");
```
Comment (tblanshard) [inline src/main/java/com/google/sps/JobStoreCenter.java:137]: Should longCurrentVcouCount be longCurrentVcpuCount or was this a deliberate choice?

**Review by tblanshard — APPROVED** [review]
LGTM :)

File: src/main/java/com/google/sps/JobStoreCenter.java (hunk 1 above)
Comment (andreeanica16) [reply src/main/java/com/google/sps/JobStoreCenter.java:137]: It was a spelling mistake. Thanks for the heads-up!

Comment (andreeanica16) [reply src/main/java/com/google/sps/JobStoreCenter.java]: Fixed here https://github.com/googleinterns/step229-2020/pull/51/commits/170f36f5dab55c3e0ab89771c1d333da18599053

**Review by tblanshard — APPROVED** [review]
LGTM :)

**Pull request merged**